tests/apollo/test_skvbc_network_partitioning.py

```diff
@@ -97,17 +95,17 @@ async def test_single_vc_primary_isolated(self, bft_network, tracker):
             await adversary.interfere()
             expected_next_primary = 1
 
-            await self._send_random_writes(skvbc)
+            await self._send_random_writes(tracker.skvbc)
```

Comment:
I don't understand this change. Why do you need to use the `skvbc` object from within the `tracker`?

Reply:
If you want to track, you just create an `skvbc` object with a `tracker` inside. If not, you create an `skvbc` object without a `tracker`. It should be that simple.

CC: @cjn @KTeofanidis

Reply:
You do understand how absurd it sounds to have a `tracker` with a reference to `skvbc` which in turn has a reference to the `tracker`, right?

Reply:
Yes True Teo.

Reply:
I'd suggest to create the `skvbc` object inside each test, passing the `tracker` as a parameter:
```python
skvbc = kvbc.SimpleKVBCProtocol(bft_network, tracker)
```
The decorator's role is to create the `tracker`, and then you pass it on to the constructor of `skvbc`.

Reply:
First case, If tracker is passed, one extra skvbc instance is created in tracker, it is not used much.

Second case, In case of no tracker, we create create skvbc. This is the current behavior.

With the current changes, to solve the duplicate skvbc, i cross referenced tracker and skvbc. This looks absurd. The second case is kept as it is.

If creation of one extra skvbc instance is fine in tracker, then I can revert this logic to old.

Reply:
Yes, I think it's fine to have the extra skvbc instance inside the tracker. Let's revert this particular change. The rest looks fine at a high level, I'd suggest @cjn to take a more detailed look next week. I'll try to find some time for it too.

Thanks Tarun!

Reply:
Sure, Thanks Teo. I will revert this particular change.